ml-agents/mlagents/trainers/policy/policy.py

```diff
@@ -114,14 +109,10 @@ def make_empty_previous_action(self, num_agents: int) -> np.ndarray:
         )
 
     def save_previous_action(
-        self, agent_ids: List[str], action_dict: Dict[str, np.ndarray]
+        self, agent_ids: List[str], action_tuple: ActionTuple
     ) -> None:
-        if action_dict is None or "discrete_action" not in action_dict:
-            return
         for index, agent_id in enumerate(agent_ids):
-            self.previous_action_dict[agent_id] = action_dict["discrete_action"][
-                index, :
-            ]
+            self.previous_action_dict[agent_id] = action_tuple.discrete[index, :]
```

Comment:
It feels really weird to me to store only the discrete action here. I suppose we could also create a new ActionTuple for each agent and store them separately